api/v1alpha1/catalogentry_types.go

```diff
@@ -0,0 +1,108 @@
+/*
+Copyright 2022 The KCP Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	kcpv1alpha1 "github.com/kcp-dev/kcp/pkg/apis/apis/v1alpha1"
+	conditionsv1alpha1 "github.com/kcp-dev/kcp/pkg/apis/third_party/conditions/apis/conditions/v1alpha1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	// InternalAPIBindingEntryLabelKey is the label key on an APIBinding with the
+	// base62(sha224(<workspace>:<exportName>)) as value to filter bindings by catalog entry.
+	InternalAPIBindingEntryLabelKey = "internal.catalog.kcp.dev/entry"
+)
+
+// These are valid conditions of CatalogEntry.
+const (
+	// CatalogEntryValid is a condition for CatalogEntry that reflects the validity
+	// of the referenced APIExport.
+	CatalogEntryValid conditionsv1alpha1.ConditionType = "CatalogEntryValid"
+	// CatalogEntryInvalidReferenceReason is a reason for the CatalogEntryValid
+	// condition of APIBinding that the referenced CatalogEntry reference is invalid.
+	CatalogEntryInvalidReferenceReason = "CatalogEntryInvalidReference"
+)
+
+//+kubebuilder:object:root=true
+//+kubebuilder:subresource:status
```

Comment:
are these automatically cluster-scoped?

Reply:
No. Default is namespaced. If you want cluster-scoped, you need to add this:
```go
//+kubebuilder:resource:scope=Cluster
```

Reply:
I think we want these to be cluster scoped?

Reply:
If we want to use workspace to organize these entries then they shouldn't be cluster-scoped.

Reply:
Changed the scoping to cluster given there is no need for namespace separation for entries.